File: backend/app/schemas/chapter.py
"""章节相关 Pydantic 模式"""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class ChapterBase(BaseModel):
    """章节基础模式"""

    title: str = Field(..., min_length=1, max_length=200, description="章节标题")
    content: Optional[str] = Field(None, description="章节内容")
    summary: Optional[str] = Field(None, description="章节摘要")
    order_index: Optional[int] = Field(0, description="排序索引")
    parent_id: Optional[int] = Field(None, description="父章节 ID")


class ChapterCreate(ChapterBase):
    """创建章节模式"""

    paper_id: int = Field(..., description="所属论文 ID")


class ChapterUpdate(BaseModel):
    """更新章节模式"""

    title: Optional[str] = Field(None, min_length=1, max_length=200)
    content: Optional[str] = None
    summary: Optional[str] = None
    order_index: Optional[int] = None
    parent_id: Optional[int] = None


class ChapterResponse(ChapterBase):
    """章节响应模式"""

    id: int
    paper_id: int
    word_count: int = 0
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class ChapterGenerateRequest(BaseModel):
    """AI 撰写章节请求模式"""

    word_count: Optional[int] = Field(1000, ge=200, le=8000, description="目标字数")
    style: Optional[str] = Field("formal", description="写作风格")
    outline_points: Optional[list] = Field(None, description="内容要点")
    use_references: Optional[bool] = Field(True, description="是否结合参考文献")


class ChapterPolishRequest(BaseModel):
    """AI 润色章节请求模式"""

    focus: Optional[str] = Field("all", description="润色重点：language / logic / format / all")

File: frontend/src/api/chapter.ts
import request from './request'

/** 章节 */
export interface Chapter {
  id: number
  paper_id: number
  parent_id: number | null
  title: string
  content: string
  summary: string
  order_index: number
  word_count: number
  created_at: string
  updated_at: string
}

/** 创建章节参数 */
export interface ChapterCreateData {
  paper_id: number
  title: string
  content?: string
  summary?: string
  order_index?: number
  parent_id?: number | null
}

/** 更新章节参数 */
export interface ChapterUpdateData {
  title?: string
  content?: string
  summary?: string
  order_index?: number
  parent_id?: number | null
}

/** 章节生成参数 */
export interface ChapterGenerateOptions {
  word_count?: number
  style?: string
  outline_points?: string[]
  use_references?: boolean
}

/** 润色侧重点 */
export type PolishFocus = 'language' | 'logic' | 'format' | 'all'

/** 获取某论文的所有章节 */
export function getChaptersByPaper(paperId: number) {
  return request.get<any, Chapter[]>(`/chapters/paper/${paperId}`)
}

/** 创建章节 */
export function createChapter(data: ChapterCreateData) {
  return request.post<any, Chapter>(`/chapters/`, data)
}

/** 获取章节详情 */
export function getChapter(id: number) {
  return request.get<any, Chapter>(`/chapters/${id}`)
}

/** 更新章节 */
export function updateChapter(id: number, data: ChapterUpdateData) {
  return request.put<any, Chapter>(`/chapters/${id}`, data)
}

/** 删除章节 */
export function deleteChapter(id: number) {
  return request.delete<any, void>(`/chapters/${id}`)
}

/** AI 撰写章节 */
export function generateChapter(id: number, options?: ChapterGenerateOptions) {
  return request.post<any, Chapter>(`/chapters/${id}/generate`, options || {})
}

/** AI 润色章节 */
export function polishChapter(id: number, focus: PolishFocus) {
  return request.post<any, Chapter>(`/chapters/${id}/polish`, { focus })
}